src/sentry/search/events/fields.py

```diff
@@ -1840,13 +1840,51 @@ class QueryFields(QueryBase):
     """Field logic for a snql query"""
 
     def resolve_select(self, selected_columns: List[str]) -> None:
+        project_key = None
+        # If project is requested, we need to map ids to their names since snuba only has ids
+        if "project" in selected_columns:
+            selected_columns.remove("project")
+            project_key = "project"
+        # since project.name is more specific, if both are included use project.name instead of project
+        if PROJECT_NAME_ALIAS in selected_columns:
+            selected_columns.remove(PROJECT_NAME_ALIAS)
+            project_key = PROJECT_NAME_ALIAS
+
         for field in selected_columns:
             if field.strip() == "":
                 continue
             resolved_field = self.resolve_field(field)
             if isinstance(resolved_field, SnqlColumn) and resolved_field not in self.columns:
                 self.columns.append(resolved_field)
 
+        if project_key:
+            self.columns.append(self.project_slug_transform(project_key))
+
+    def project_slug_transform(self, project_key: str) -> SnqlFunction:
+        """When project is a selected column we need to create a transform to turn them back into slugs"""
+        project_ids = {
+            project_id
+            for project_id in self.params.get("project_id", [])
+            if isinstance(project_id, int)
+        }
+
+        # Try to reduce the size of the transform by using any existing conditions on projects
+        if len(self.projects_to_filter) > 0:
+            project_ids &= self.projects_to_filter
+
+        projects = Project.objects.filter(id__in=project_ids).values("slug", "id")
+
+        return SnqlFunction(
+            "transform",
+            [
+                self.column("project.id"),
+                [project["id"] for project in projects],
+                [project["slug"] for project in projects],
+                "",
+            ],
+            project_key,
+        )
```

Comment:
This is so much cleaner than the previous transform, no `array` functions, no random toString on the project.id 😭